### PR TITLE
Add developer guide

### PR DESCRIPTION
**Pull request by FummiTaksi**
Developer guides how to write your own handler. The README has link to the developer guide.

### CODE REVIEW

File: DEVGUIDE.md [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+        self.assertEqual(test_suite['name'], 'Locust test suite, failure percentage 70')
+```
+
+All 8 tests should pass. Now we have completed an LocustHandler with unit tests which test the most important functionalities.
+
+
+
+## How to package your project
+
+Let's package our project in the same virtual environment . Add necessary files defined in this [tutorial](https://packaging.python.org/) to `locustenv` folder and install `setuptools` and `wheel` to your virtualenv with `pip install`. Add following line yo your `setup.py` file in the `setuptools.setup() object`:
```
Comment (Tattoo) [inline DEVGUIDE.md:614]: `setuptools` and `wheel` are part of Python standard library ie. installed by default. Please remove sentence about installing them.

File: DEVGUIDE.md [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+```
+locusthandler.locusthandler:
+  handler: LocustHandler
+  keyword: run_locust
+  tags: oxygen-locusthandler
+  failure_percentage: 20
+```
+
+
+ Next let's run the robot test case to make sure that it works.  Copy `test.robot` and `locustfile.py` files to `packagenv/` folder, make the following changes to the variables in `test.robot`:
```
Comment (Tattoo) [inline DEVGUIDE.md:657]: the copying of files need a bit clarification because I was confused about why are we copying locusthandler's code since we have already installed it in this place. Maybe something like: "let's make our Locust tests and Robot Framework tests that run them available in our environment." Or something

File: DEVGUIDE.md [hunk 3]
```diff
@@ -0,0 +1,771 @@
...
+````
+
+and all 3 tests should pass.
+
+
+### Configuring LocustHandler to Oxygen
+
+Let's open the python interpreter from the `locustenv` directory and check that we can import the locusthandler:
+
+```
```
[the remark this answers is not in the document]
Comment (FummiTaksi) [reply DEVGUIDE.md:172]: Commited a change for this

File: DEVGUIDE.md [hunk 4]
```diff
@@ -0,0 +1,771 @@
...
+            'suites': [],
+            'tests': test_cases,
+            }
+            return test_suite
+```
+
+Let's update the tests to match the current functionality. Let's start by defining new data set in `locustenv/locusthandler/resources/requests.csv`:
+
+```
+"Type","Name","Request Count","Failure Count","Median Response Time","Average Response Time","Min Response Time","Max Response Time","Average Content Size","Requests/s","Failures/s","50%","66%","75%","80%","90%","95%","98%","99%","99.9%","99.99%","99.999%","100%"
```
[the remark this answers is not in the document]
Comment (FummiTaksi) [reply DEVGUIDE.md:353]: The new data set has different values in the third data row ( 30 requests, 3 failures) to test the edge case of the `failure_precentage` in `test_pass_is_true_when_failure_request_percentage_is_ten` test case . This is documented now better.

File: DEVGUIDE.md [hunk 5]
```diff
@@ -0,0 +1,771 @@
...
+    def test_pass_is_true_when_failure_request_percentage_is_zero(self):
+        self.assertEqual(self.test_suite['tests'][0]['keywords'][0]['pass'], True)
+
+    def test_pass_is_true_when_failure_request_percentage_is_ten(self):
+        self.assertEqual(self.test_suite['tests'][2]['keywords'][0]['pass'], True)
+
+    def test_pass_is_false_when_failure_request_percentage_is_above_ten(self):
+        self.assertEqual(self.test_suite['tests'][1]['keywords'][0]['pass'], False)
+
+    def test_failure_percentage_is_ten_by_default(self):
```
[the remark this answers is not in the document]
Comment (FummiTaksi) [reply DEVGUIDE.md:388]: `test_failure_percentage_is_ten_by_default` tests that the default value of `_get_treshold_failure_percentage` is 10.
`test_pass_is_true_when_failure_request_is_default_value`  ( Notice new name) tests, that the `pass` value is `True` when the amount of failed requests is exactly on the limit, which happens to be 10.

So there is two test cases because another tests that the default value is correct, and another tests that in edge case the value of `pass` will be `True`

File: DEVGUIDE.md (hunk 2 above)
Comment (FummiTaksi) [reply DEVGUIDE.md]: Committed a change for this

File: DEVGUIDE.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Tattoo) [reply DEVGUIDE.md:388]: But given the `pass` wouldn't be `True` if the percentage is not 10 (or whatever the default value is), the test case also tests this invariant.  So the more specific "this is the default" test case is redundant. 

Generally, having these kinds of more specific redundant tests are only good for documenting important details (to explicitly draw attention to them), but in this case, there is no grand point why it is important that the threshold is 10 -- you say it in the guide yourself that this was just flight of fancy. Therefore, we have a redundant test that underlines a design decision that is of no importance. 

In fact, we explicitly implement two different ways for the user to define the threshold themselves because our guess would be just arbitrary. So therefore "shielding" this design decision doesen't, to me, make much sense.

File: DEVGUIDE.md (hunk 1 above)
Comment (FummiTaksi) [reply DEVGUIDE.md]: `wheel` needs to be installed

File: DEVGUIDE.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (FummiTaksi) [reply DEVGUIDE.md]: It is true that the default value will be tested anyways in the test case which tests the edge case of the treshold. Removed the `test_failure_percentage_is_ten_by_default`

File: DEVGUIDE.md [hunk 6]
```diff
@@ -0,0 +1,722 @@
...
+           'robotframework-oxygen>=0.1',
+           'locust>=1.1',
+      ],
+```
+
+so that oxygen including it's dependencies and locust will be installed when your handler is installed. Next you can run following command from `locustenv` folder:
+
+```
+pip install wheel
+python setup.py sdist bdist_wheel
```
Comment (Tattoo) [inline DEVGUIDE.md:587]: we don't need sdist in this command, it creates a tar.gz which in the age of wheel is redundant

Comment (FummiTaksi) [reply DEVGUIDE.md]: Committed a change for this

File: DEVGUIDE.md [hunk 7]
```diff
@@ -0,0 +1,722 @@
...
+
+```
+*** Settings ***
+Library   oxygen.OxygenLibrary
+Library   OperatingSystem
+
+*** Variables ***
+${STATS_FILE}       ${CURDIR}/../example_stats.csv
+${FAILURE_FILE}     ${CURDIR}/../example_failures.csv
+${HISTORY_FILE}     ${CURDIR}/../example_stats_history.csv
```
Comment (Tattoo) [inline DEVGUIDE.md:236]: These variables are never used anywhere, maybe remove them from the example?

Comment (FummiTaksi) [reply DEVGUIDE.md:244]: They are generated from the test so I think it is best to delete them also before running the test incase the developer running the test wants to do something with them

Comment (Tattoo) [reply DEVGUIDE.md:244]: Ah, you are completely right

**Comment by Tattoo** [comment]
lgtm, outstanding work
